library/src/main/kotlin/com/chuckerteam/chucker/internal/support/SpanTextUtil.kt

```diff
@@ -0,0 +1,132 @@
+package com.chuckerteam.chucker.internal.support
+
+import android.graphics.Color
+import android.text.SpannableStringBuilder
+import android.text.Spanned
+import android.text.style.ForegroundColorSpan
+import androidx.core.text.isDigitsOnly
+import com.google.gson.JsonElement
+import com.google.gson.JsonParser
+import com.google.gson.JsonSyntaxException
+
+
+public class SpanTextUtil {
+    public companion object {
+        private val JSON_KEY_COLOR = Color.parseColor("#8B0057")
+        private val JSON_STRING_VALUE_COLOR = Color.parseColor("#2F00FF")
+        private val JSON_DIGIT_AND_NULL_VALUE_COLOR = Color.parseColor("#E84B31")
+        private val JSON_SIGN_ELEMENTS_COLOR = Color.parseColor("#474747")
+
+        public fun spanJson(input: CharSequence): SpannableStringBuilder {
+            val jsonElement = try {
+                JsonParser.parseString(input.toString())
+            } catch (e: JsonSyntaxException) {
+                Logger.warn("Json structure is invalid so it can not be formatted", e)
+                return SpannableStringBuilder.valueOf(input)
+            }
+            val sb = SpannableStringBuilder()
+            printifyRecursive(jsonElement, StringBuilder(""), sb)
+            return sb
+        }
+
+        private fun printifyRecursive(
+            transformedJson: JsonElement,
+            currentIndent: StringBuilder,
+            sb: SpannableStringBuilder
+        ) {
+            val indent = StringBuilder(currentIndent)
+            if (transformedJson.isJsonArray) {
+                printifyJsonArray(sb, indent, transformedJson)
+            }
+            if (transformedJson.isJsonObject) {
+                printifyJsonObject(sb, indent, transformedJson)
+            }
+        }
+
+        private fun printifyJsonArray(
+            sb: SpannableStringBuilder,
+            indent: StringBuilder,
+            transformedJson: JsonElement
+        ) {
+            if (transformedJson.asJsonArray.isEmpty) {
+                sb.appendWithColor(
+                    "[]",
+                    JSON_SIGN_ELEMENTS_COLOR
+                )
+                return
+            }
+            sb.appendWithColor("$indent[\n", JSON_SIGN_ELEMENTS_COLOR)
```

Comment:
nit: for the sake of consistency
```suggestion
            sb.appendWithColor("${indent}[\n", JSON_SIGN_ELEMENTS_COLOR)